src/ir/ty.rs

```diff
@@ -387,6 +387,18 @@ impl Type {
             _ => false,
         }
     }
+
+    pub(crate) fn needs_wasm32_abi_hack(&self, ctx: &BindgenContext) -> bool {
+        ctx.is_target_wasm32_not_emscripten() &&
+            match self.kind {
+                TypeKind::ResolvedTypeRef(id) |
+                TypeKind::Alias(id) => {
+                    ctx.resolve_type(id).needs_wasm32_abi_hack(ctx)
+                }
+                TypeKind::Comp(_) => true,
```

Comment:
not sure what to do here. compinfo::layout() is none, compinfo.fields() -> field::layout() is none, etc. How do I figure out the size and shape of this comp?

Reply:
You want `self.layout(ctx)`, `compinfo::layout()` is just a fallback if we can't get the right size from llvm.